framework/audio/internal/audiobuffer.h

```diff
@@ -0,0 +1,56 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_BUFFER_H
+#define MU_AUDIO_BUFFER_H
+
+#include <vector>
+#include <memory>
+#include "iaudiosource.h"
+
+namespace mu {
+namespace audio {
+class AudioBuffer
+{
+    const static unsigned int DEFAULT_SIZE = 4096;
+    const static unsigned int MIN_SAMPLELAG = 2048;
+    const static unsigned int MAX_SAMPLELAG = 3072;
+    const static unsigned int FILL_SAMPLES = 1024;
+
+public:
+    AudioBuffer(unsigned int streamsPerSample = 2, unsigned int size = DEFAULT_SIZE);
+
+    void setSource(std::shared_ptr<IAudioSource> source);
+    void fillup();
+
+    void push(const float* source, int sampleCount);
+    void pop(float* dest, int sampleCount);
+
+    unsigned int sampleLag() const;
+
+private:
+    unsigned int m_streamsPerSample;
```

Comment:
You should use the default initialization with c++ primitives

Reply:
everything initilized in the constructor:
```
AudioBuffer::AudioBuffer(unsigned int streamsPerSample, unsigned int size)
    : m_streamsPerSample(streamsPerSample), m_writepointer(0), m_readpointer(0), m_data(), m_source(nullptr)
```

Reply:
This is old-style initialization, better to initialize primitives with a something default value after the declaration

Reply:
fixed